zebra/zebra_rib.c

```diff
@@ -969,9 +973,22 @@ static unsigned nexthop_active_check(struct route_node *rn,
 
 	memset(&nexthop->rmap_src.ipv6, 0, sizeof(union g_addr));
 
+	r_nexthop = nexthop;
+	/* Get the first valid ifindex in case of recersive route
+	 */
+	while ((r_nexthop->ifindex == 0) && (r_nexthop->resolved))
```

Comment:
Why this change to look at different nexthops?  What issue are you solving?

Reply:
This code is required for the recursive routes and VRF leak scenarios. 
Scenario : 
/** configure a route map with interface **/
configure terminal
route-map rmap1 permit 1
match interface ens224
end

/** apply the route map on static routes in zebra **/

ip protocol static route-map rmap1 

/** configure a static route **/
c t
ip route 99.0.0.1 255.255.255.255 ens192
end

/** configure a recurcive route */
c t
ip route 9.9.9.9 255.255.255.255 99.0.0.1
end


Problem statement :
9.9.9.9 is discarded by the Route-map

RCA :
re(99.0.0.1)--->
              nexthop (vrf->id = 0, ifindex=0)
              |
              resolved-> NULL

re(9.9.9.9)---->
      nexthop (vrf->id = 0, ifindex=0)
      |
      resolved-> nexthop(vrf_id = 0, ifindex=ens224)

Routemap Comparision :
(interface(ens224),nexthop->vrf_id)==nexthop->ifindex

Here if we used nexthop ifindex then routemap returns the NO_MATCH.

Solution:
We need to find out the resolved nexthop. The same thing is applicable for VRF leak.

Reply:
1) General nexthop/route-map behavior has always been to match on the non-recursed nexthop.  This is a major behavioral change from people's expectation and we should not be sneaking this in with this other feature.  It should be in a different PR ( not that I think that this behavior is correct mind you )
2) If you do want to submit another PR then you must provide a better methodology of doing this that actually would respect all the different nexthops that it resolves to instead of just what ever one you find(as that there is no guarantee of any order here ) before I would even consider it.  I would also expect how you plan to migrate people from old behavior to new behavior in a logical manner to be listed as part of the PR.